src/compute/src/server.rs

```diff
@@ -675,18 +682,19 @@ impl<'w, A: Allocate + 'static> Worker<'w, A> {
                 // Sink tokens should be retained for retained dataflows, and dropped for dropped
                 // dataflows.
                 //
-                // Dropping the tokens of active subscribes makes them place `DroppedAt` responses
-                // into the subscribe response buffer. We drop that buffer in the next step, which
-                // ensures that we don't send out `DroppedAt` responses for subscribes dropped
-                // during reconciliation.
+                // Dropping the tokens of active subscribe and copy-tos makes them place
+                // `DroppedAt` responses into the respective response buffer. We drop those buffers
+                // in the next step, which ensures that we don't send out `DroppedAt` responses for
+                // subscribe/copy-tos dropped during reconciliation.
                 if !retained {
                     collection.sink_token = None;
                 }
             }
 
-            // We must drop the subscribe response buffer as it is global across all subscribes.
-            // If it were broken out by `GlobalId` then we could drop only those of dataflows we drop.
+            // We must drop the response buffers as they are global across all subscribe/copy-tos.
+            // If they were broken out by `GlobalId` then we could drop only those of dataflows we drop.
             compute_state.subscribe_response_buffer = Rc::new(RefCell::new(Vec::new()));
+            compute_state.copy_to_response_buffer = Rc::new(RefCell::new(Vec::new()));
```

Comment:
Should we convert the response buffers to something like a `BTreeMap<GlobalId, Rc<RefCell<...>>>`? Not part of this PR though!